config/config.go

```diff
@@ -324,13 +334,32 @@ func (c *configStruct) Logout(endpointURL string) {
 	}
 
 	if element, ok := c.Endpoints[ep]; ok {
+		element.RefreshToken = ""
 		element.Token = ""
 		element.Scheme = ""
 	}
 
 	WriteToFile()
 }
 
+// AuthHeaderGetter returns a function that can get the auth header for a given endpoint that the client can use.
+// The returned function will attempt to refresh the token in case the scheme is Bearer and the token is expired.
+func (c *configStruct) AuthHeaderGetter(endpoint string, overridingToken string) func() (authheader string, err error) {
+	return func() (string, error) {
+		token := c.ChooseToken(endpoint, overridingToken)
+		scheme := c.ChooseScheme(endpoint, overridingToken)
+
+		// If the scheme is Bearer, first verify that the token is expired.
+		// And if it is expired, then try to refresh it.
```

Comment:
The next PR would then implement what this comment is saying. But first I'd like to get approval on the approach taken in this PR.